File: src/main/java/org/gga/graph/search/bfs/AbstractBfsVisitor.java
package org.gga.graph.search.bfs;

import org.gga.graph.Graph;
import org.gga.graph.Edge;

/**
 * @author mike
 */
public class AbstractBfsVisitor implements BfsVisitor {
    public void initializeVertex(int v, Graph g) {
    }

    public void discoverVertex(int v, Graph g) {
    }

    public void examineVertex(int v, Graph g) {
    }

    public void finishVertex(int v, Graph g) {
    }

    public void examineEdge(Edge e, Graph g) {
    }

    public void treeEdge(Edge e, Graph g) {
    }

    public void nonTreeEdge(Edge e, Graph g) {
    }

    public void greyTarget(Edge e, Graph g) {
    }

    public void blackTarget(Edge e, Graph g) {
    }
}

File: src/main/scala/org/gga/graph/search/bfs/AbstractBfsVisitor.scala
package org.gga.graph.search.bfs

import org.gga.graph.Edge
import org.gga.graph.Graph

/**
 * @author mike
 */
trait AbstractBfsVisitor extends BfsVisitor {
  override def initializeVertex(v: Int, g: Graph) {
  }

  override def discoverVertex(v: Int, g: Graph) {
  }

  override def examineVertex(v: Int, g: Graph) {
  }

  override def finishVertex(v: Int, g: Graph) {
  }

  override def examineEdge(e: Edge, g: Graph) {
  }

  override def treeEdge(e: Edge, g: Graph) {
  }

  override def nonTreeEdge(e: Edge, g: Graph) {
  }

  override def grayTarget(e: Edge, g: Graph) {
  }

  override def blackTarget(e: Edge, g: Graph) {
  }
}